src/server/reset_to_version.rs

```diff
@@ -225,34 +231,50 @@ impl ResetToVersionManager {
             warn!("A reset-to-version process is already in progress! Wait until it finish first.");
             self.wait();
         }
-        *self.worker_handle.borrow_mut() = Some(std::thread::Builder::new()
-            .name("reset_to_version".to_string())
-            .spawn_wrapper(move || {
-                tikv_util::thread_group::set_properties(props);
-                tikv_alloc::add_thread_memory_accessor();
-
-                while worker.process_next_batch(BATCH_SIZE, &mut wb).expect("reset_to_version failed when removing invalid writes") {
-                }
-                *worker.state.lock()
-                        .expect("failed to lock `ResetToVersionWorker::state` in `ResetToVersionWorker::process_next_batch`")
-                    = ResetToVersionState::RemovingLock { scanned: 0 };
-                while worker.process_next_batch_lock(BATCH_SIZE, &mut wb).expect("reset_to_version failed when removing invalid locks") {
-                }
-                *worker.state.lock()
-                        .expect("failed to lock `ResetToVersionWorker::state` in `ResetToVersionWorker::process_next_batch_lock`")
-                    = ResetToVersionState::Done;
-                info!("Reset to version done!");
-                tikv_alloc::remove_thread_memory_accessor();
-            })
-            .expect("failed to spawn reset_to_version thread"));
+        *self.worker_handle.borrow_mut() = Some(
+            thread::Builder::new()
+                .name("reset_to_version".to_string())
+                .spawn_wrapper(move || {
+                    tikv_util::thread_group::set_properties(props);
+                    tikv_alloc::add_thread_memory_accessor();
+                    // Delete the write/default data in batch first.
+                    loop {
+                        match worker.process_next_batch(&mut wb) {
+                            Ok(has_more) => {
+                                if !has_more {
+                                    break;
+                                }
+                            }
+                            Err(e) => {
+                                error!("reset_to_version failed when removing from write/default cfs"; "error" => ?e);
+                                break;
+                            }
+                        }
+                    }
+                    // Delete the lock data in batch then.
+                    loop {
+                        match worker.process_next_batch_lock(&mut wb) {
```

Comment:
I don't know the purpose of dividing it into two stages. @longfangsong If an error happened in the midst of `process_next_batch`, would it leave an incorrect state?